contract-sdk/src/caller.rs

```diff
@@ -0,0 +1,90 @@
+use alloc::string::String;
+use alloc::vec::Vec;
+use core::cell::{Ref, RefCell, RefMut};
+use core::ops::{Deref, DerefMut};
+
+use crate::Identity;
+use crate::{Blob, StructuredBlob};
+use bincode::{Decode, Encode};
+
+use serde::{Deserialize, Serialize};
+#[derive(Debug, Serialize, Deserialize, Clone, Encode, Decode, Default)]
+pub struct ExecutionState {
```

Comment:
I find this naming not very relevant, it's not really a state imho, maybe more a ctx ?

Reply:
renamed